src/rt-app_taskgroups.c

```diff
@@ -373,7 +402,14 @@ static int cgroup_attach_task(char *name)
 	FILE *tasks;
 	size_t size;
 
-	file = strcmp(name, "/") ? "/tasks" : "tasks";
+	if (ctrl.version == 1) {
+		file = strcmp(name, "/") ? "/tasks" : "tasks";
+	}
+
+	if (ctrl.version == 2) {
+		file = strcmp(name, "/") ? "/cgroup.procs" : "cgroup.procs";
```

Comment:
Notice this moves whole process (threadgroup in contrast to `tasks` that works with threads).
If this should support also thread granularity, `cgroup.type` should be used too (see e.g. cgroups(7)).

Reply:
I have changed it to use "cgroup.threads". Plz review again.